sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -237,9 +228,7 @@ private static void registerAndRunFramework(
 
         // The framework ID is not available when we're being started for the first time.
         Optional<Protos.FrameworkID> optionalFrameworkId = stateStore.fetchFrameworkId();
-        if (optionalFrameworkId.isPresent()) {
-            fwkInfoBuilder.setId(optionalFrameworkId.get());
-        }
+        optionalFrameworkId.ifPresent(fwkInfoBuilder::setId);
```

Comment:
In the uninstall case, we should behave differently when optionalFrameworkId is missing: If it's missing then it implies ZK is missing, likely due to a previously completed uninstall.

In this scenario, the scheduler would ideally start in a no-op mode where it's not doing anything other than advertising a completed plan state.